multitenant: add more log messages around RU limiting

RaduBerinde opened the pull request:
#### tenantcostserver: add some verbose logs

This commit adds some verbose logs describing the token bucket
requests that are happening.

Release note: None

#### tenantcostserver: add logs when reconfiguring RUs for a tenant

This commit adds a log message when a tenant token bucket is
reconfigured.

Release note: None

#### tenantcostclient: add trace event for delayed requests

This change adds a trace event for any request that has to wait for
more than 1 second in the RU limiting subsystem.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76386)
<!-- Reviewable:end -->

RaduBerinde committed:
tenantcostserver: add some verbose logs

This commit adds some verbose logs describing the token bucket
requests that are happening.

Release note: None

RaduBerinde committed:
tenantcostserver: add logs when reconfiguring RUs for a tenant

This commit adds a log message when a tenant token bucket is
reconfigured.

Release note: None

RaduBerinde committed:
tenantcostclient: add trace event for delayed requests

This change adds a trace event for any request that has to wait for
more than 1 second in the RU limiting subsystem.

Release note: None

RaduBerinde commented:
bors r+
